clients/service/src/lib.rs

```diff
@@ -67,7 +71,28 @@ impl<Config: Clone + Send + 'static, F: Fn()> ConnectionManager<Config, F> {
 	pub async fn start<S: Service<Config, InnerError>, InnerError: fmt::Display>(
 		&self,
 	) -> Result<(), Error<InnerError>> {
+		let mut restart_delay_timer_in_secs = 20; // set default to 20 seconds for restart
+		let mut time_as_of_recording = SystemTime::now();
+
 		loop {
+			let time_now = SystemTime::now();
+			let _ = time_now.duration_since(time_as_of_recording).map(|duration| {
+				// Revert the counter if the restart happened more than 30 minutes (or 1800 seconds)
+				// ago
+				if duration.as_secs() > 1800 {
+					restart_delay_timer_in_secs = 20;
+				}
+				// Increase time by 10 seconds if a restart is triggered too frequently.
+				// This waits for delayed packets to be removed in the network,
+				// even though the connection on the client side is closed.
+				// Else, these straggler packets will interfere with the new connection.
+				// https://www.rfc-editor.org/rfc/rfc793#page-22
+				else {
+					restart_delay_timer_in_secs += 10;
```

Comment:
If the retry/reconnect is executed within 1800 seconds or 30 minutes, increase the delay by 10.